starknet-rpc-test/get_block_hash_and_number.rs

```diff
@@ -1,19 +1,52 @@
 extern crate starknet_rpc_test;
 
+use starknet_accounts::SingleOwnerAccount;
+use starknet_core::chain_id;
 use starknet_ff::FieldElement;
 use starknet_providers::Provider;
-use starknet_rpc_test::{ExecutionStrategy, MadaraClient};
+use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, MINT_AMOUNT, SIGNER_PRIVATE};
+use starknet_rpc_test::utils::transfer_tokens;
+use starknet_rpc_test::{BlockCreation, ExecutionStrategy, MadaraClient};
+use starknet_signers::{LocalWallet, SigningKey};
 
 #[tokio::test]
-async fn work_ok_at_initialization() -> Result<(), anyhow::Error> {
+async fn work_ok_at_start_and_with_new_blocks() -> Result<(), anyhow::Error> {
     let madara = MadaraClient::new(ExecutionStrategy::Native).await;
     let rpc = madara.get_starknet_client();
 
     assert_eq!(
         rpc.block_hash_and_number().await?.block_hash,
-        FieldElement::from_hex_be("0x031ebd02657f940683ae7bddf19716932c56d463fc16662d14031f8635df52ad").unwrap()
+        FieldElement::from_hex_be("0x035d394a3808b432702261c2d66f1cf2f111dd0c691ad2a8246b86a5088f73fa").unwrap()
```

Comment:
I would rather have you also call `get_block` and compute the hash of the block you received.
Otherwise, we have to trust you on the fact that this is its correct hash.
Also if we change the block structure or the way we compute it's hash. Those tests will break

Reply:
There is no `compute_hash` method easily accessible on the `starknet_core::Block`?

Reply:
Unfortunately not, I suggest we let that for another PR since it might need a bit more work